### PR TITLE
Using session_device_id for user filter if unidentified

### LINKED ISSUE #7183: clicking the 'search for other sessions' icon for a non-identified session populates an invalid filter
Screenshot 2023-11-20 at 1 28 22 PM

https://discord.com/channels/1026884757667188757/1174463824288567308

clicking the button next to the identifier on the top right for a session that is not identified populates the `identifier` query with something that doesn't match. instead, we should use the `clientID` query

image

### PR DESCRIPTION
**Pull request by deltaepsilon**
## Summary

Closes #7183 

image

## How did you test this change?

Click test

## Are there any deployment considerations?

nope!

## Does this work require review from our design team?

nope!

### CODE REVIEW

File: frontend/src/pages/Player/MetadataBox/MetadataBox.tsx [hunk 1]
```diff
@@ -189,12 +189,16 @@ export const MetadataBox = React.memo(() => {
 		const displayName = getDisplayName(session)
 		const userParam = validateEmail(displayName) ? 'email' : 'identifier'
 
-		setSearchQuery((query) =>
-			buildQueryStateString({
+		setSearchQuery((query) => {
+			const params = session.identified
+				? { [`user_${userParam}`]: displayName }
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Player/MetadataBox/MetadataBox.tsx:194]: Just wanted to check if you knew of the `getDisplayNameAndField` method? Seems to have the different cases listed with the value and identifier.

**Review by SpennyNDaJets — APPROVED** [review]
Click test looks good to me! Just one non-blocking question below to make sure we are handling any edge cases

File: frontend/src/pages/Player/MetadataBox/MetadataBox.tsx (hunk 1 above)
Comment (deltaepsilon) [reply frontend/src/pages/Player/MetadataBox/MetadataBox.tsx:194]: Thanks for the heads up!

I've reviewed the function. It looks like it's doing a fall-through where it looks for name, email, identifier or fingerprint. I think this change still checks out, because the fingerprint should be universally available and maps to the `session_device_id`.
